frontend/pages/schedule/ManageSchedulePage/components/ScheduleEditorModal/ScheduleEditorModal.tsx

```diff
@@ -0,0 +1,257 @@
+import React, { useState, useCallback } from "react";
+
+import { pull } from "lodash";
+// @ts-ignore
+import FleetIcon from "components/icons/FleetIcon";
+import Modal from "components/modals/Modal";
+import Button from "components/buttons/Button";
+import InfoBanner from "components/InfoBanner/InfoBanner";
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+import { IQuery } from "interfaces/query";
+import { IGlobalScheduledQuery } from "interfaces/global_scheduled_query";
+import {
+  FREQUENCY_DROPDOWN_OPTIONS,
+  PLATFORM_OPTIONS,
+  LOGGING_TYPE_OPTIONS,
+  MIN_OSQUERY_VERSION_OPTIONS,
+} from "utilities/constants";
+
+const baseClass = "schedule-editor-modal";
+interface IScheduleEditorModalProps {
+  allQueries: IQuery[];
+  onCancel: () => void;
+  onScheduleSubmit: (formData: any) => void;
+}
+interface IFrequencyOption {
+  value: number;
+  label: string;
+}
+
+interface INoQueryOption {
+  id: number;
+  name: string;
+}
+
+const ScheduleEditorModal = ({
+  onCancel,
+  onScheduleSubmit,
+  allQueries,
+}: IScheduleEditorModalProps): JSX.Element => {
+  const [showAdvancedOptions, setShowAdvancedOptions] = useState<boolean>(
+    false
+  );
+  const [selectedQuery, setSelectedQuery] = useState<
+    IGlobalScheduledQuery | INoQueryOption
+  >();
+  const [selectedFrequency, setSelectedFrequency] = useState<number>(86400);
+  const [
+    selectedPlatformOptions,
+    setSelectedPlatformOptions,
+  ] = useState<string>("");
+  const [selectedLoggingType, setSelectedLoggingType] = useState<string>(
+    "snapshot"
+  );
+  const [
+    selectedMinOsqueryVersionOptions,
+    setSelectedMinOsqueryVersionOptions,
+  ] = useState(null);
+  const [selectedShard, setSelectedShard] = useState(null);
+
+  const createQueryDropdownOptions = () => {
+    const queryOptions = allQueries.map((q: any) => {
+      return {
+        value: String(q.id),
+        label: q.name,
+      };
+    });
+    return [...queryOptions];
```

Comment:
Can this just be 
```suggestion
    return queryOptions;
```
since `Array.prototype.map` returns a new array?